src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -224,9 +175,151 @@ public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithPartitionCount
+                    }
+                },
+            Tags = { { "aspire-resource-name", infrastructure.AspireResource.Name } }
+        };
+        infrastructure.Add(cosmosAccount);
+
+        foreach (var database in azureResource.Databases)
+        {
+            var cosmosSqlDatabase = new CosmosDBSqlDatabase(Infrastructure.NormalizeBicepIdentifier(database.Name))
+            {
+                Parent = cosmosAccount,
+                Name = database.Name,
+                Resource = new CosmosDBSqlDatabaseResourceInfo()
+                {
+                    DatabaseName = database.Name
+                }
+            };
+            infrastructure.Add(cosmosSqlDatabase);
+
+            foreach (var container in database.Containers)
+            {
+                var cosmosContainer = new CosmosDBSqlContainer(Infrastructure.NormalizeBicepIdentifier(container.Name))
+                {
+                    Parent = cosmosSqlDatabase,
+                    Name = container.Name,
+                    Resource = new CosmosDBSqlContainerResourceInfo()
+                    {
+                        ContainerName = container.Name,
+                        PartitionKey = new CosmosDBContainerPartitionKey { Paths = [container.PartitionKeyPath] }
+                    }
+                };
+                infrastructure.Add(cosmosContainer);
+            }
+        }
+
+        if (azureResource.UseAccessKeyAuthentication)
+        {
+            cosmosAccount.DisableLocalAuth = false;
+
+            var kvNameParam = new ProvisioningParameter(AzureBicepResource.KnownParameters.KeyVaultName, typeof(string));
+            infrastructure.Add(kvNameParam);
+
+            var keyVault = KeyVaultService.FromExisting("keyVault");
+            keyVault.Name = kvNameParam;
+            infrastructure.Add(keyVault);
+
+            var secret = new KeyVaultSecret("connectionString")
+            {
+                Parent = keyVault,
+                Name = "connectionString",
+                Properties = new SecretProperties
+                {
+                    Value = BicepFunction.Interpolate($"AccountEndpoint={cosmosAccount.DocumentEndpoint};AccountKey={cosmosAccount.GetKeys().PrimaryMasterKey}")
+                }
+            };
+            infrastructure.Add(secret);
+        }
+        else
+        {
+            cosmosAccount.DisableLocalAuth = true;
+
+            var principalIdParameter = new ProvisioningParameter(AzureBicepResource.KnownParameters.PrincipalId, typeof(string));
+            infrastructure.Add(principalIdParameter);
+
+            cosmosAccount.CreateRoleAssignment(CosmosDBBuiltInRole.DocumentDBAccountContributor, RoleManagementPrincipalType.ServicePrincipal, principalIdParameter);
+
+            infrastructure.Add(new ProvisioningOutput("connectionString", typeof(string))
+            {
+                Value = BicepFunction.Interpolate($"https://{cosmosAccount.Name}.document.azure.com")
```

Comment:
This doesn't seem like the right thing to do.
```suggestion
                Value = cosmosAccount.DocumentEndpoint
```

Reply:
Done and tested